### PR TITLE
feat: bounded balance

### PR DESCRIPTION
**Pull request by victoraugustolls**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,24 +16,24 @@ GOLANGCI_LINT_VERSION=1.41.1
 .PHONY: setup
 setup:
 	@echo "==> Setup: Tidying modules"
-	go mod tidy
+	# go mod tidy
```
Comment (thalissonfelipe) [inline Makefile:19]: Qual o motivo dos comentários?

File: app/domain/usecases/get_account_balance_test.go [hunk 2]
```diff
@@ -85,8 +93,37 @@ func TestLedgerUseCase_GetAccountBalance_Synthetic(t *testing.T) {
...
+		assert.NoError(t, err)
+
+		queryBalance := vos.NewSyntheticAccountBalance(account, 20)
+		mockedRepository := &mocks.RepositoryMock{
+			GetBoundedAccountBalanceFunc: func(_ context.Context, _ vos.Account, _, _ time.Time) (vos.AccountBalance, error) {
+				return queryBalance, nil
+			},
+		}
+
+		nr, _ := newrelic.NewApplication()
```
Comment (thalissonfelipe) [inline app/domain/usecases/get_account_balance_test.go:116]: Is the second argument an error? If so, wouldn't it be better to add an assert.NoError(t, err)?

File: app/gateways/db/postgres/get_bounded_account_balance.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	const operation = "Repository.GetBoundedAccountBalance"
+
+	query, args := buildBoundedBalanceQuery(acc, start, end)
+
+	defer newrelic.NewDatastoreSegment(ctx, collection, operation, query).End()
+
+	var balance int
+
+	err := r.db.QueryRow(ctx, query, args...).Scan(&balance)
+
```
Comment (thalissonfelipe) [inline app/gateways/db/postgres/get_bounded_account_balance.go:42]: Can we remove this blank line?

File: app/domain/usecases/get_account_balance_test.go (hunk 2 above)
Comment (victoraugustolls) [reply app/domain/usecases/get_account_balance_test.go:116]: I wold prefer to this in a separate PR! Many of our tests are doing this, so a PR reviewing all of our tests ir needed!

File: app/gateways/db/postgres/get_bounded_account_balance.go (hunk 3 above)
Comment (victoraugustolls) [reply app/gateways/db/postgres/get_bounded_account_balance.go]: NIT, as this should be fixed in another PR by reviewing our linters. But done!

File: Makefile (hunk 1 above)
Comment (victoraugustolls) [reply Makefile]: Done

**Review by thalissonfelipe — APPROVED** [review]

**Review by Streppel — APPROVED** [review]

**Pull request merged**